dev-games/godot/godot-3.2.3.ebuild

```diff
@@ -0,0 +1,136 @@
+
+RDEPEND="
+	>=app-arch/bzip2-1.0.6-r6
+	>=app-arch/lz4-0_p120
+	>=app-arch/xz-utils-5.0.8
+	>=app-arch/zstd-1.4.4
+	>=dev-libs/json-c-0.11-r1
+	>=media-libs/alsa-lib-1.0.28
+	>=media-libs/flac-1.3.1-r1
+	>=media-libs/libpng-1.6.16:0=
+	>=media-libs/libsndfile-1.0.25-r1
+	>=media-libs/mesa-10.2.8[gles2]
+	>=net-libs/libasyncns-0.8-r3
+	>=sys-apps/attr-2.4.47-r1
+	>=sys-apps/tcp-wrappers-7.6.22-r1
+	>=sys-apps/util-linux-2.25.2-r2
+	>=sys-libs/gdbm-1.11
+	>=sys-libs/glibc-2.20-r2
+	>=sys-libs/libcap-2.22-r2
+	>=sys-libs/zlib-1.2.8-r1
+	>=x11-libs/libX11-1.6.2
+	>=x11-libs/libXcursor-1.1.14
+	>=x11-libs/libXinerama-1.1.3
+	media-libs/alsa-lib
+	dev-libs/libpcre2[pcre32]
+	media-libs/libvpx
+	virtual/glu
+	virtual/opengl
+	bullet? ( >=sci-physics/bullet-2.89 )
+	enet? ( net-libs/enet:= )
+	freetype? ( >=media-libs/freetype-2.5.3-r1:2 )
+	mbedtls? ( net-libs/mbedtls )
+	ogg? ( >=media-libs/libogg-1.3.1 )
+	opus? (
+		media-libs/opus
+		media-libs/opusfile
+	)
+	pulseaudio? ( >=media-sound/pulseaudio-5.0-r7 )
+	theora? ( media-libs/libtheora )
+	udev? ( virtual/udev )
+	upnp? ( net-libs/miniupnpc )
+	vorbis? ( >=media-libs/libvorbis-1.3.4 )
+	webp? ( media-libs/libwebp )
+"
+DEPEND="${RDEPEND}"
+BDEPEND="virtual/pkgconfig"
+
+src_configure() {
+	strip-unsupported-flags
+
+	myesconsargs=(
+		CC="$(tc-getCC)"
+		CXX="$(tc-getCXX)"
+	)
+	# Remove builtin third-party packages, link with system ones instead
+	myesconsargs+=(
+		builtin_bullet=no
+		builtin_enet=no
+		builtin_freetype=no
+		builtin_libogg=no
+		builtin_libpng=no
+		builtin_libtheora=no
+		builtin_libvorbis=no
+		builtin_libvpx=no
+		builtin_libwebp=no
+		builtin_mbedtls=no
+		builtin_miniupnpc=no
+		builtin_opus=no
+		builtin_pcre2=no
+		builtin_zlib=no
+		builtin_zstd=no
+	)
+	myesconsargs+=(
+		# Mono bindings requires MSBuild which is only available on Windows
+		module_mono_enabled=no
+		module_bullet_enabled=$(usex bullet)
+		module_enet_enabled=$(usex enet)
+		module_freetype_enabled=$(usex freetype)
+		module_mbedtls_enabled=$(usex mbedtls)
+		module_ogg_enabled=$(usex ogg)
+		module_opus_enabled=$(usex opus)
+		module_theora_enabled=$(usex theora)
+		module_upnp_enabled=$(usex upnp)
+		module_vorbis_enabled=$(usex vorbis)
+		module_webp_enabled=$(usex webp)
+	)
+	# Misc options
+	myesconsargs+=(
+		platform=x11
+		progress=yes
+		tools=yes
+		verbose=yes
+		pulseaudio=$(usex pulseaudio)
+		target=$(usex debug debug release_debug)
+		udev=$(usex udev)
+		use_lto=$(usex lto)
+	)
+}
+
+src_compile() {
+	escons "${myesconsargs[@]}"
+}
+
+src_install() {
+	local godot_binary="${PN}.x11.opt.tools"
+
+	if [[ "${ARCH}" == "amd64" ]]; then
+		godot_binary="${godot_binary}.64"
+	elif [[ "${ARCH}" == "x86" ]]; then
+		godot_binary="${godot_binary}.32"
+	fi
+
+	newicon icon.svg ${PN}.svg
+	dobin bin/${godot_binary}
+	dodoc AUTHORS.md CHANGELOG.md DONORS.md README.md
+
+	make_desktop_entry ${godot_binary} Godot
```

Comment:
I see a `dist` directory upstream: https://github.com/godotengine/godot/tree/master/misc/dist/linux

You might be able to use the [`xdg` eclass](https://devmanual.gentoo.org/eclass-reference/xdg.eclass/) to automatically take care of the installation for this desktop entry by calling `xdg_pkg_postinst` instead.